go/tools/gazelle/rules/resolve.go

```diff
@@ -56,3 +55,36 @@ func (l label) String() string {
 	}
 	return fmt.Sprintf("%s//%s:%s", repo, l.pkg, l.name)
 }
+
+func NewLabelResolver(c *config.Config) LabelResolver {
+	var e LabelResolver
+	switch c.DepMode {
+	case config.ExternalMode:
+		e = newExternalResolver()
+	case config.VendorMode:
+		e = vendoredResolver{}
+	}
+
+	return &unifiedResolver{
```

Comment:
Kind of. This change means `rules.Generator` is instantiated for each package, since it now has package-specific state. That may be expanded if we support other directives or inferences in existing BUILD files. I wanted to keep a single instance of `rules.LabelResolver` though, so I needed to make it public so `main.run` could pass it to `rules.NewGenerator`.

I took the opportunity to do a little refactoring when I extracted this constructor from `rules.NewGenerator`. I'd like to pull the resolver into a new package later though before I add new functionality.